### PR TITLE
feat: `get_my_proposal_review_summary` endpoint

### PR DESCRIPTION
**Pull request by ilbertt**
Added the `get_my_proposal_review_summary` endpoint, together with integration tests.

Blocked by https://github.com/CodeGov-org/codegov-website/pull/110.

### CODE REVIEW

File: src/backend/impl/src/repositories/types/proposal_review_commit.rs [hunk 1]
```diff
@@ -11,13 +11,38 @@ use super::{CommitSha, DateTime, ProposalReviewId, UserId, Uuid};
...
+    pub matches_description: Option<bool>,
+    pub comment: Option<String>,
+    pub highlights: Vec<String>,
+}
+
+impl Display for ReviewedCommitState {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut commit_details = format!(
+            "Matches description: {}",
+            self.matches_description.unwrap_or(false),
```
Comment (nathanosdev) [inline src/backend/impl/src/repositories/types/proposal_review_commit.rs]: Maybe it's better to put here "Unanswered" or something similar to make sure it's clear the difference between the reviewer not answering and them explicitly saying that the commit does not match it's description.

File: src/backend/impl/src/services/proposal_review_service.rs [hunk 2]
```diff
@@ -652,6 +644,156 @@ impl<
...
+            "# Proposal {}\n\n",
+            proposal.nervous_system.proposal_id()
+        ));
+    }
+    // info
+    {
+        md_content.push_str(&format!("Vote: {}\n", proposal_review.vote));
+        md_content.push_str(&format!(
+            "Hashes match: {}\n",
+            proposal_review.build_reproduced.unwrap_or(false)
```
Comment (nathanosdev) [inline src/backend/impl/src/services/proposal_review_service.rs]: Similarly here, maybe it's better to be explicit and put "Unanswered" or something similar.

File: src/backend/impl/src/services/proposal_review_service.rs [hunk 3]
```diff
@@ -652,6 +644,156 @@ impl<
...
+        ));
+        md_content.push_str(&format!(
+            "All reviewed commits match their descriptions: {}\n",
+            reviewed_commits.iter().all(|(_, commit)| commit
+                .reviewed_state()
+                .and_then(|state| state.matches_description)
+                .unwrap_or(false))
```
Comment (nathanosdev) [inline src/backend/impl/src/services/proposal_review_service.rs]: This one is a little trickier for the case that only some (or even none) of the commits have answers with regards to whether or not they match the description. Maybe something like "Only partially answered (see individual reports below)" when there's only some not answered and "Unanswered" when none of them are.

**Review by nathanosdev — APPROVED** [review]

File: src/backend/impl/src/repositories/types/proposal_review_commit.rs (hunk 1 above)
Comment (ilbertt) [reply src/backend/impl/src/repositories/types/proposal_review_commit.rs]: Fixed

File: src/backend/impl/src/services/proposal_review_service.rs (hunk 2 above)
Comment (ilbertt) [reply src/backend/impl/src/services/proposal_review_service.rs]: Fixed

File: src/backend/impl/src/services/proposal_review_service.rs (hunk 3 above)
Comment (ilbertt) [reply src/backend/impl/src/services/proposal_review_service.rs]: Fixed

**Pull request merged**